gui/src/2D/controls/control.ts

```diff
@@ -1617,24 +1617,32 @@ export class Control {
     }
 
     protected _evaluateClippingState(parentMeasure: Measure) {
+        let evaluatedMeasure = new Measure(0, 0, 0, 0);
+        this._currentMeasure.transformToRef(this._transformMatrix, evaluatedMeasure);
+
+        let evaluatedParentMeasure = new Measure(0, 0, 0, 0);
```

Comment:
We should not allocate as much, you could keep the instance of measure in cache and reuse it. Also, Nit: it should be a const._evaluateClippingState